File: dynamic-pricing-business-rules-engine/frontend/src/types/pricingHistory.ts
export interface PricingHistoryRule {
  rule_id: number;
  rule_name: string;
  action_type: string;
  discount_amount: number;
}

export interface PricingHistory {
  calculation_id: number;
  product_id: number;
  customer_id: number;
  quantity: number;
  location: string | null;
  promotional_code: string | null;
  original_price: number;
  discount_amount: number;
  tax_amount: number;
  final_price: number;
  calculated_at: string;
  applied_rules: PricingHistoryRule[];
}

export interface PricingHistoryListResponse {
  items: PricingHistory[];
  total: number;
  skip: number;
  limit: number;
}

File: dynamic-pricing-business-rules-engine/backend/app/schemas/pricing_history.py
from datetime import datetime
from decimal import Decimal

from pydantic import BaseModel, ConfigDict


class PricingHistoryRuleResponse(BaseModel):
    rule_id: int
    rule_name: str
    action_type: str
    discount_amount: Decimal

    model_config = ConfigDict(from_attributes=True)


class PricingHistoryResponse(BaseModel):
    calculation_id: int
    product_id: int
    customer_id: int
    quantity: int
    location: str | None
    promotional_code: str | None
    original_price: Decimal
    discount_amount: Decimal
    tax_amount: Decimal
    final_price: Decimal
    calculated_at: datetime
    applied_rules: list[PricingHistoryRuleResponse] = []


class PricingHistoryListResponse(BaseModel):
    items: list[PricingHistoryResponse]
    total: int
    skip: int
    limit: int
